### ISSUE TITLE
Function that can find all available polygones to a point within a certain radius

### ISSUE THREAD

**Issue by jingjunL** [r-spatial/sf#1779]
Hello everyone,

I would like to achieve the below function via SF  and I am not sure how to programme it. Basically, assuming there is a point A c(3000, 50000) and a bunch of polygons group B around this point, what I want to do is to first create a polygon that using point A as a circle centre and has a radius of 200 meters (namely Polygone C). Then, I would like to find all available polygons in group B that pass through polygon C.

Maybe an image will make my clarification more clear. 
![IMG_0101](https://user-images.githubusercontent.com/63421908/131817832-6b8c7a41-bf9b-4bd5-8d0b-596dd0aa258c.jpg)

Basically, as you can see in the image, the yellow polygon with a "selected" is the only one selected and the other two polygons that are either out or within polygon C are not selected.

I am wondering if you are aware of any function that can achieve my problem? Thanks very much for your help in advance :)

**Comment by tim-salabim** [comment]
There's no one function to do this, but the workflow would roughly be like this:

1. `st_buffer` around the point
2.  `st_cast` the buffer to `LINESTRING`
3. `st_intersects` between the linestring from 2. and your polygons

HTH

As a general note, this question is not suitable for github issues as it seeks general advice rather than being a dedicated bug report or feature request. R-sig-geo mailing list or stackoverflow would be better places for such questions.

**Comment by edzer** [comment]
For your goal, it is may not be needed to create the circular polygon; `st_is_within_distance()` may be enough; see also https://r-spatial.github.io/sf/articles/sf7.html#st_buffer-or-st_is_within_distance

**Comment by jingjunL** [comment]
> For your goal, it is may not be needed to create the circular polygon; `st_is_within_distance()` may be enough; see also https://r-spatial.github.io/sf/articles/sf7.html#st_buffer-or-st_is_within_distance

Hello Edzer:

From my understanding of "within", isn't `st_is_within_distance` count all polygons that are less than 2km away from `point A` in my case (which means the "unselected" polygon within the circle in my image will also be selected)? Or if I have misunderstood this? That would be much handier if my desired function can be achieved with this simple formula :)

**Comment by edzer** [comment]
Ah, I see; yes, I misread.

**Comment by jingjunL** [comment]
> Ah, I see; yes, I misread.

No problem :) Thanks very much and I will close this issue!
